packages/react-instantsearch/src/core/createHistoryStateManager.js

```diff
@@ -37,9 +37,13 @@ function applyStateToLocation(location, state, knownKeys) {
       query,
     };
   }
+
   return {
     ...location,
-    search: query ? `?${qs.stringify(query, {sort: alphabeticalSort})}` : '',
+    search:
+      query && Object.keys(query).length > 0 ?
+        `?${qs.stringify(query, {sort: alphabeticalSort, arrayFormat: 'repeat'})}`
```

Comment:
This allow having categories=product&categories=test instead of categories[0]=product&categories[1]=test (coupled with some changes in getCurrentRefinement in most list widgest